charts/generic-service/templates/ingress_cp.yaml

```diff
@@ -25,7 +25,7 @@ metadata:
     nginx.ingress.kubernetes.io/modsecurity-snippet: {{ tpl (toYaml .Values.ingress.modsecurity_snippet) . | nindent 6 }}
   {{- end }}
 spec:
-  ingressClassName: {{- if .Values.ingress.modsecurity_enabled }} "modsec" {{- else }} "default" {{- end }}
+  ingressClassName: {{- if .Values.ingress.className }} {{ .Values.ingress.className }} {{- else if .Values.ingress.modsecurity_enabled }} "modsec" {{- else }} "default" {{- end }}
```

Comment:
Do you wanted to, you could add a test value to `hmpps-helm-charts/charts/generic-service/ci/test-values.yaml`